docs/src/pages/theming/theming.web.mdx

```diff
@@ -106,6 +106,45 @@ const myTheme = {
 };
 ```
 
+#### Defining component tokens
+
+Amplify UI should follow a consistent pattern when defining tokens for a component's states and variations. This is helpful for discovering what tokens are available for theming different aspects of a component. Amplify UI uses the following pattern:
+
+```javascript
+component[variant][_state][child];
+```
+
+Following this pattern, our token definition for the component would follow this general structure. **Note**: Amplify UI prefixes states with an underscore, `_`, to help distinguish state names from variation names.
+
+```javascript
+export const component = {
```

Comment:
Feels like it might be more useful if we provide a real example from the codebase?

Reply:
Updated to include an abridged example (didn't want to include the whole file) from the Button component.